modulus/datapipes/climate/utils/invariant.py

```diff
@@ -0,0 +1,119 @@
+from abc import ABC, abstractmethod
+from typing import Tuple, List
+
+import numpy as np
+import xarray as xr
+
+
+def latlon_grid(
+    bounds: Tuple[Tuple[float, float], Tuple[float, float]] = (
+        (90, -90),
+        (0, 360),
+    ),
+    shape: Tuple[int, int] = (1440,721)    
+) -> np.ndarray :
+    """Infer latitude and longitude coordinates from bounds and data shape."""
+
+    # get latitudes and longitudes from data shape
+    lat = np.linspace(*bounds[0], shape[0], dtype=np.float32)
+
+    # does longitude wrap around the globe?
+    lon_wraparound = ((bounds[1][0] % 360) == (bounds[1][1] % 360))
+    if lon_wraparound:
+        # treat differently from lat due to wrap-around
+        lon = np.linspace(*bounds[1], shape[1]+1, dtype=np.float32)[:-1]
+    else:
+        lon = np.linspace(*bounds[1], shape[1], dtype=np.float32)
+
+    return np.meshgrid(lat, lon, indexing='ij')
+
+
+class Invariant(ABC):
+    @abstractmethod
+    def __call__(self, latlon: np.ndarray):
+        pass
```

Comment:
I would appreciate more docstring on this class and what it is intended for.

Reply:
I'll add a docstring (and there should be a type hint for the return value as well) to this tomorrow. It's just there to specify the interface that the `Invariant` subclasses are supposed to follow.